### PR TITLE
dev

### PR DESCRIPTION
**Pull request by maxlandon**
- **Update README**
- **Update carapace dep**
- **Add a system shell command to optional commands**
- **Add an exit command**
- **Format**

### CODE REVIEW

File: commands/exit.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// readline receives an io.EOF error, which is returned with CtrlD.
+func exitCtrlD() {
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Print("Confirm exit (Y/y): ")
+
+	text, _ := reader.ReadString('\n')
+	answer := strings.TrimSpace(text)
+
+	if (answer == "Y") || (answer == "y") {
```
Comment (ccoVeille) [inline commands/exit.go:37]:
```suggestion
	if strings.EqualFold(answer, "y")
```

File: commands/shell.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			if len(args) == 0 {
+				return errors.New("command requires one or more arguments")
+			}
+
+			path, err := exec.LookPath(args[0])
+			if err != nil {
+				return err
+			}
+
+			shellCmd := exec.Command(path, args[1:]...)
```
Comment (ccoVeille) [inline commands/shell.go:29]: Won't it fail if there is only one argument?

Shouldn't the test on len look for 2, not 1

See https://github.com/reeflective/console/pull/58#discussion_r1844996446

File: commands/shell.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+// ExecuteShell returns a cobra command to execute a line through the system shell.
+// This uses the os/exec package to execute the command. The default command name is `!`.
+func ExecuteShell() *cobra.Command {
+	shellCmd := &cobra.Command{
+		Use:                "!",
+		Short:              "Execute the remaining arguments with system shell",
+		DisableFlagParsing: true,
+		RunE: func(_ *cobra.Command, args []string) error {
+			if len(args) == 0 {
```
Comment (ccoVeille) [inline commands/shell.go:20]:
```suggestion
			if len(args) < 2 {
```

See

https://github.com/reeflective/console/pull/58#discussion_r1844996541

https://github.com/reeflective/console/pull/58#discussion_r1844996301

File: commands/shell.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+				return errors.New("command requires one or more arguments")
+			}
+
+			path, err := exec.LookPath(args[0])
```
Comment (ccoVeille) [inline commands/shell.go:24]: This will panic if args is empty

**Comment by maxlandon** [comment]
Integrated your fixes/remarks in the last release 0.1.20

**Comment by ccoVeille** [comment]
> Integrated your fixes/remarks in the last release 0.1.20

For anyone curious, take a look at

- #59

**Pull request merged**